typescript/lib/constructor_calls.ts

```diff
@@ -0,0 +1,211 @@
+    const result: Map<string, FileGenerationOptions> = new Map();
+    for (let fileId in filesToGenerateSet) {
+        result[fileId] = new FileGenerationOptions();
+        result[fileId].setFileTypeToGenerate(FileGenerationOptions.FileTypeToGenerate.FILE);
+    }
+
+    return result;
+}
+
+export function newGetLoadStaticFilesArgs(serviceId: ServiceID, staticFilesToCopyStringSet: Map<String, boolean>): LoadStaticFilesArgs {
+    const result: LoadStaticFilesArgs = new LoadStaticFilesArgs();
+    result.setServiceId(String(serviceId));
+    const staticFilesMap = result.getStaticFilesMap();
+    for (let staticFildID in staticFilesToCopyStringSet) {
+        staticFilesMap.set(staticFildID, staticFilesToCopyStringSet[staticFildID]);
+    }
+
+    return result;
+} 
+
+// // ====================================================================================================
+// //                                    Network Context
+// // ====================================================================================================
+
+export function newGetLoadLambdaArgs(lambdaId: LambdaID, image: string, serializedParams: string): LoadLambdaArgs {
+    const result: LoadLambdaArgs = new LoadLambdaArgs();
+    result.setLambdaId(String(lambdaId));
+    result.setContainerImage(image);
+    result.setSerializedParams(serializedParams);
+
+    return result;
+}
+
+export function newGetLambdaInfoArgs(lambdaId: LambdaID): GetLambdaInfoArgs {
+    const result: GetLambdaInfoArgs = new GetLambdaInfoArgs();
+    result.setLambdaId(String(lambdaId)); //TODO - String(lambdaId) VS <string>lambdaId for type assertions (also seen "as")
+
+    return result;
+}
+
+export function newRegisterStaticFilesArgs(strSet: Map<string, boolean>): RegisterStaticFilesArgs {
+    const result: RegisterStaticFilesArgs = new RegisterStaticFilesArgs();
+    const staticFilesSetMap: Map<string, boolean> = result.getStaticFilesSetMap();
+    for (let staticFileID in strSet) {
+        staticFilesSetMap.set(staticFileID, strSet[staticFileID]);
+    }
+
+    return result;
+}
+
+export function newRegisterFilesArtifactsArgs(filesArtifactIdStrsToUrls: Map<string, string>): RegisterFilesArtifactsArgs {
+    const result: RegisterFilesArtifactsArgs = new RegisterFilesArtifactsArgs();
+    const filesArtifactUrlsMap = result.getFilesArtifactUrlsMap();
+    for (let fileArtificactID in filesArtifactUrlsMap) {
+        filesArtifactUrlsMap.set(fileArtificactID, filesArtifactUrlsMap[fileArtificactID]);
+    }
+    return result;
+}
+
+export function newRegisterServiceArgs(serviceId: ServiceID, partitionId: PartitionID): RegisterServiceArgs {
+    const result: RegisterServiceArgs = new RegisterServiceArgs();
+    result.setServiceId(<string>serviceId);
+    result.setPartitionId(<string>partitionId);
+
+    return result;
+}
+
+export function newStartServiceArgs(
```

Comment:
Omg having this "constructor" function is so much better than needing to inline this wherever you wanted this